### PR TITLE
test: cover untested branches in render_detail.py (#1058)

### LINKED ISSUE #1058: [aw][test audit] render_detail.py: untested branches in _build_event_details (SESSION_SHUTDOWN), _render_shutdown_cycles (None t
[Content truncated due to length]
## Root Cause

Three private helpers in `src/copilot_usage/render_detail.py` have branches that are never directly asserted against in `tests/copilot_usage/test_render_detail.py`. Existing `TestBuildEventDetailsToolRequests`, `TestBuildEventDetailsUserMessage`, and `TestSafeEventData` only exercise the `USER_MESSAGE`, `ASSISTANT_MESSAGE`, and `TOOL_EXECUTION_COMPLETE` arms of `_build_event_details`. The `SESSION_SHUTDOWN` arm (lines 151–157), the `None`-timestamp path of `_render_shutdown_cycles` (line 244), and the `max_len ≤ 0` guard of `_truncate` (lines 55–62) are all unreachable via any current assertion.

---

## Gap 1 — `_build_event_details`: SESSION_SHUTDOWN arm never directly tested

**Source:** `src/copilot_usage/render_detail.py` lines 151–157

```python
case EventType.SESSION_SHUTDOWN:
    if (data := _safe_event_data(ev, ev.as_session_shutdown)) is None:
        return ""
    return f"type={data.shutdownType}" if data.shutdownType else ""
```

Three sub-cases need direct coverage:

| Sub-case | Input | Expected output |
|---|---|---|
| Non-empty `shutdownType` | `data={"shutdownType": "routine", ...}` | `"type=routine"` |
| Empty `shutdownType` | `data={"shutdownType": "", ...}` | `""` |
| Malformed data (None from `_safe_event_data`) | `data={"shutdownType": 99}` (fails Pydantic) | `""` |

**Spec – add a `TestBuildEventDetailsSessionShutdown` class:**

```python
class TestBuildEventDetailsSessionShutdown:
    def test_non_empty_shutdown_type(self) -> None:
        ev = SessionEvent(
            type=EventType.SESSION_SHUTDOWN,
            data={
                "shutdownType": "routine",
                "totalPremiumRequests": 0,
                "totalApiDurationMs": 0,
                "modelMetrics": {},
            },
        )
        assert _build_event_details(ev) == "type=routine"

    def test_empty_shutdown_type(self) -> None:
        ev = SessionEvent(
            type=EventType.SESSION_SHUTDOWN,
            data={
                "shutdownType": "",
                "totalPremiumRequests": 0,
                "totalApiDurationMs": 0,
                "modelMetrics": {},
            },
        )
        assert _build_event_details(ev) == ""

    def test_malformed_data_returns_empty(self) -> None:
        ev = SessionEvent(
            type=EventType.SESSION_SHUTDOWN,
            data={"shutdownType": 99},  # int triggers ValidationError
        )
        assert _build_event_details(ev) == ""
```

---

## Gap 2 — `_render_shutdown_cycles`: `ts = None` path renders `"—"`

**Source:** `src/copilot_usage/render_detail.py` line 244

```python
date_str = ts.strftime("%Y-%m-%d %H:%M") if ts else "—"
```

`SessionSummary.shutdown_cycles` is typed `list[tuple[datetime | None, SessionShutdownData]]`; a `None` timestamp is therefore a valid and supported input. No existing test passes `(None, sd)` to `_render_shutdown_cycles`.

**Spec – add a test to the existing `TestRenderShutdownCycles` class (or a new focused class):**

```python
def test_none_timestamp_renders_dash(self) -> None:
    """A shutdown cycle with ts=None must display '—' in the Date column."""
    sd = SessionShutdownData(
        shutdownType="routine",
        totalPremiumRequests=1,
        totalApiDurationMs=500,
        modelMetrics={},
    )
    summary = SessionSummary(
        session_id="no-ts",
        shutdown_cycles=[(None, sd)],
    )
    buf = io.StringIO()
    console = Console(file=buf, force_terminal=True, width=120)
    _render_shutdown_cycles(summary, target_console=console)
    output = _strip_ansi(buf.getvalue())
    assert "—" in output
```

---

## Gap 3 — `_truncate`: `max_len ≤ 0` defensive guard uncovered

**Source:** `src/copilot_usage/render_detail.py` lines 55–62

```python
def _truncate(text: str, max_len: int = MAX_CONTENT_LEN) -> str:
    if max_len <= 0:
        return ""
    ...
```

All current callers pass `max_len=60` or the default `MAX_CONTENT_LEN=80`, so the guard is never reached. The branch is dead code, but it is still an untested path that could silently change semantics if `max_len` usage were ever widened.

**Spec – add parametrized edge-case tests near the existing truncation tests:**

```python
`@pytest`.mark.parametrize("max_len", [0, -1, -100])
def test_truncate_non_positive_max_len_returns_empty(max_len: int) -> None:
    """_truncate must return '' for any max_len ≤ 0."""
    assert _truncate("hello", max_len) == ""
```

---

## Acceptance Criteria

- [ ] `TestBuildEventDetailsSessionShutdown` class added with three tests (non-empty type, empty type, malformed data)
- [ ] At least one test for `_render_shutdown_cycles` with `ts=None` asserting `"—"` in output
- [ ] Parametrized test covering `max_len ∈ {0, -1, -100}` for `_truncate`
- [ ] All new tests pass with `pytest tests/copilot_usage/test_render_detail.py`
- [ ] No existing tests broken

> Generated by [Test Suite Analysis](https://github.com/microsasa/cli-tools/actions/runs/24815517813/agentic_workflow) · ● 8.7M · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+is%3Aissue+%22gh-aw-workflow-call-id%3A+microsasa%2Fcli-tools%2Ftest-analysis%22&type=issues)

### PR DESCRIPTION
**Pull request by microsasa**
Closes #1058

Adds direct test coverage for three untested branches in `src/copilot_usage/render_detail.py`:

### Gap 1 — `_build_event_details`: SESSION_SHUTDOWN arm
New `TestBuildEventDetailsSessionShutdown` class with three tests:
- **`test_non_empty_shutdown_type`** — verifies `"type=routine"` output
- **`test_empty_shutdown_type`** — verifies empty string output
- **`test_malformed_data_returns_empty`** — int `shutdownType` triggers `ValidationError` → `""`

### Gap 2 — `_render_shutdown_cycles`: `ts=None` path
New `TestRenderShutdownCyclesNoneTimestamp` class:
- **`test_none_timestamp_renders_dash`** — `(None, sd)` tuple renders `"—"` in Date column

### Gap 3 — `_truncate`: `max_len ≤ 0` guard
Parametrized test covering `max_len ∈ {0, -1, -100}`:
- **`test_truncate_non_positive_max_len_returns_empty`** — all return `""`

### Notes
- No source changes — test-only additions
- All tests follow existing patterns in the file (same imports, helpers, style)
- Environment could not run `make check` locally (network-blocked sandbox; `uv` not available), but CI will validate

> [!WARNING]
> <details>
> <summary><strong>⚠️ Firewall blocked 2 domains</strong></summary>
>
> The following domains were blocked by the firewall during workflow execution:
>
> - `astral.sh`
> - `pypi.org`
>
> To allow these domains, add them to the `network.allowed` list in your workflow frontmatter:
>
> ```yaml
> network:
>   allowed:
>     - defaults
>     - "astral.sh"
>     - "pypi.org"
> ```
>
> See [Network Configuration](https://github.github.com/gh-aw/reference/network/) for more information.
>
> </details>

> Generated by [Issue Implementer](https://github.com/microsasa/cli-tools/actions/runs/24824695584/agentic_workflow) · ● 14.9M · [◷](https://github.com/search?q=repo%3Amicrosasa%2Fcli-tools+%22gh-aw-workflow-id%3A+issue-implementer%22&type=pullrequests)

### CODE REVIEW

File: tests/copilot_usage/test_render_detail.py [hunk 1]
```diff
@@ -957,3 +958,84 @@ def test_build_event_details_returns_empty_for_unrecognized_type() -> None:
...
+            totalPremiumRequests=1,
+            totalApiDurationMs=500,
+            modelMetrics={},
+        )
+        summary = SessionSummary(
+            session_id="no-ts",
+            shutdown_cycles=[(None, sd)],
+        )
+        buf = io.StringIO()
+        console = Console(file=buf, force_terminal=True, width=120)
```
[the remark this answers is not in the document]
Comment (microsasa) [reply tests/copilot_usage/test_render_detail.py]: Good catch! Replaced all 5 inline `StringIO`/`Console` setups in `TestRenderCodeChanges` with `_buf_console()` to keep console configuration consistent and eliminate duplication. No other inline setups remain in the file.

> [!WARNING]
> <details>
> <summary><strong>⚠️ Firewall blocked 1 domain</strong></summary>
>
> The following domain was blocked by the firewall during workflow execution:
>
> - `pypi.org`
>
> To allow these domains, add them to the `network.allowed` list in your workflow frontmatter:
>
> ```yaml
> network:
>   allowed:
>     - defaults
>     - "pypi.org"
> ```
>
> See [Network Configuration](https://github.github.com/gh-aw/reference/network/) for more information.
>
> </details>


> Generated by [Review Responder](https://github.com/microsasa/cli-tools/actions/runs/24825599598/agentic_workflow) · ● 3.3M
